.github/workflows/main.yml

```diff
@@ -0,0 +1,124 @@
+            template: kubernetes
+            var-files: common,kvm,linux,ubuntu-focal,kubernetes,kubernetes_1_25
+          - name: kubernetes-1-25-jammy
+            template: kubernetes
+            var-files: common,kvm,linux,ubuntu-jammy,kubernetes,kubernetes_1_25
+          - name: kubernetes-1-26-focal
+            template: kubernetes
+            var-files: common,kvm,linux,ubuntu-focal,kubernetes,kubernetes_1_26
+          - name: kubernetes-1-26-jammy
+            template: kubernetes
+            var-files: common,kvm,linux,ubuntu-jammy,kubernetes,kubernetes_1_26
+          - name: kubernetes-1-27-focal
+            template: kubernetes
+            var-files: common,kvm,linux,ubuntu-focal,kubernetes,kubernetes_1_27
+          - name: kubernetes-1-27-jammy
+            template: kubernetes
+            var-files: common,kvm,linux,ubuntu-jammy,kubernetes,kubernetes_1_27
+    name: ${{ matrix.name }}
+    steps:
+      - name: Check out the repository
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Write OpenStack credentials
+        run: echo "$CLOUDS_YAML_B64" | base64 -d > ./clouds.yaml
+        env:
+          CLOUDS_YAML_B64: ${{ secrets.CLOUDS_YAML_B64 }}
+
+      - name: Set up Packer environment
+        run: ./bin/setup
+        env:
+          PACKER_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build image
+        id: build-image
+        run: ./bin/build-image
+        env:
+          OS_CLOUD: openstack
+          PACKER_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          ENVIRONMENT: arcus
+          PACKER_TEMPLATE: ${{ matrix.template }}
+          ENV_VAR_FILES: ${{ matrix.var-files }}
+
+      - name: Publish image
+        id: publish-image
+        run: ./bin/publish-image
+        env:
+          OS_CLOUD: openstack
+          ENVIRONMENT: arcus
+          ENV_VAR_FILES: ${{ matrix.var-files }}
+          IMAGE_ID: ${{ steps.build-image.outputs.image-id }}
+          S3_ACCESS_KEY: ${{ secrets.S3_ACCESS_KEY }}
+          S3_SECRET_KEY: ${{ secrets.S3_SECRET_KEY }}
+
+      - name: Write matrix outputs
+        uses: cloudposse/github-action-matrix-outputs-write@main
+        with:
+          matrix-step-name: ${{ github.job }}
+          matrix-key: ${{ matrix.name }}
+          outputs: |-
+            name: ${{ steps.publish-image.outputs.image-name }}
+            url: ${{ steps.publish-image.outputs.image-url }}
+            checksum: ${{ steps.publish-image.outputs.image-checksum }}
+            manifest-extra: ${{ steps.build-image.outputs.manifest-extra }}
+
+  publish_manifest:
+    needs: [build_images]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the repository
+        uses: actions/checkout@v3
+
+      - name: Read matrix outputs
+        id: matrix-outputs
+        uses: cloudposse/github-action-matrix-outputs-read@main
+        with:
+          matrix-step-name: build_images
+
+      - name: Write outputs
```

Comment:
https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions suggests, amongst other things, pinning the version to a full length commit.

Reply:
I agree